resolver/resolver_test.go

```diff
@@ -108,40 +123,53 @@ func TestRecurivePathResolution(t *testing.T) {
 
 func TestResolveToLastNode_NoUnnecessaryFetching(t *testing.T) {
 	ctx := context.Background()
-	dagService := dagmock.Mock()
+	bsrv := mockBlockService()
 
 	a := randNode()
 	b := randNode()
 
 	err := a.AddNodeLink("child", b)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
-	err = dagService.Add(ctx, a)
-	if err != nil {
-		t.Fatal(err)
-	}
+	err = bsrv.AddBlock(a)
+	require.NoError(t, err)
 
 	aKey := a.Cid()
 
 	segments := []string{aKey.String(), "child"}
 	p, err := path.FromSegments("/ipfs/", segments...)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
+
+	resolver := resolver.NewBasicResolver(bsrv)
+	resolver.FetchConfig.NodeReifier = unixfsnode.Reify
 
-	resolver := resolver.NewBasicResolver(dagService)
 	resolvedCID, remainingPath, err := resolver.ResolveToLastNode(ctx, p)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
-	if len(remainingPath) > 0 {
-		t.Fatal("cannot have remaining path")
-	}
+	require.Equal(t, len(remainingPath), 0, "cannot have remaining path")
+	require.Equal(t, b.Cid(), resolvedCID)
+}
 
-	if !resolvedCID.Equals(b.Cid()) {
-		t.Fatal("resolved to the wrong CID")
-	}
+func TestPathRemainder(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	bsrv := mockBlockService()
+
+	nd, err := ipldcbor.FromJSON(strings.NewReader(`{"foo": {"bar": "baz"}}`), math.MaxUint64, -1)
```

Comment:
Not the fault of this library, but `math.MaxUint64, -1` seems like a non-intuitive convention to me and especially so when the function doesn't have any comments to clarify behavior.

Given that the only reason the go-ipld-cbor library is being brought into this library at all is for this test, would it be better to use an ipld-prime based codec library (e.g. for dag-cbor or dag-json)?

Reply:
removed